keps/sig-cli/2590-kubectl-subresource/README.md

```diff
@@ -0,0 +1,393 @@
+involves using `kubectl --raw`. Patching subresources using kubectl is not possible at all and 
+requires using curl directly. This method is very cumbersome and not user-friendly.
+
+This KEP adds subresources as a first class option in kubectl to allow users to work with the API
+in a generic fashion.
+
+### Goals
+
+- Add a new flag `--subresource=[subresource-name]` to get, apply, patch, edit
+  and replace kubectl commands to allow fetching and updating `status` and `scale` subresources for all resources
+  (built-in and CRs) that support these subresources.
+- Display pretty printed table columns for the status (uses same columns as the main resource) and scale subresources.
+
+
+### Non-Goals
+
+- Support subresources other than `status` and `scale`.
+- Allow updates to the status subresource if a controller is running against the main
+  resource that reconciles `.status` part of the resource.
+- Allow specifying `additionalPrinterColumns` for CRDs for the status subresource.
+
+## Proposal
+
+kubectl commands like get, apply, patch, edit and replace will now contain a
+new flag `--subresource=[subresource-name]` which will allow fetching and updating
+`status` and `scale` subresources for all API resources.
+
+When the `--subresource=status` flag is used against built-in API types, 
+the `.status` field of the resource is _not_ updated since the built-in controllers run against
+the resource and reconcile the `.status` part. This also holds true for custom resources if a 
+custom controller reconciles the `.status` part of the CR. It ensures that `.status` of 
+built-in API types will never store arbitrary data.
+
+If `--subresource` flag is used for a resource that doesn't support the subresource, 
+a `NotFound` error will be returned.
+
+### Examples
+
+#### get
+
+```shell
+# for built-in types
+$ kubectl get deployment nginx-deployment --subresource=status
+NAME               READY   UP-TO-DATE   AVAILABLE   AGE
+nginx-deployment   3/3     3            3           43s
+
+$ kubectl get deployment nginx-deployment --subresource=scale
+NAME               DESIREDREPLICAS   AVAILABLEREPLICAS
+nginx-deployment   3                 3
+
+# for CRDS
+$ kubectl get crontab cron--subresource=status
+NAME   SPEC          REPLICAS   AGE
+cron   * * * * */5   3          4m52s
+
+$ kubectl get vmset vmset-1 --subresource=scale
+NAME   DESIREDREPLICAS   AVAILABLEREPLICAS
+cron   3                 0
+```
+
+Invalid subresources:
+
+```shell
+$ kubectl get pod nginx-deployment-66b6c48dd5-dv6gl --subresource=logs
+error: --subresource must be one of [status scale], not "logs"
+
+$ kubectl get pod nginx-deployment-66b6c48dd5-dv6gl --subresource=scale
+Error from server (NotFound): the server could not find the requested resource
+```
+
+#### patch
+
+```shell
+# For built-in types
+# update spec.replicas through scale subresource
+$ kubectl patch deployment nginx-deployment --subresource='scale' --type='merge' -p '{"spec":{"replicas":2}}'
```

Comment:
What's the actual benefit of doing that though?

Reply:
The ability to modify the scale subresource was added to be consistent with the similar behavior provided to the status subresrouce.

@nikhita if you had any specific uses cases for modifying scale, we should document them in the KEP.